File: RecruiterLite/Models/Company.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace RecruiterLite.Models;

public class Company : BaseEntity
{
    public string CompanyName { get; set; }
    public string PhoneNumber { get; set; }
    public string StreetAddress { get; set; }
    public string PostCode { get; set; }
    public string County { get; set; }
    public string Country { get; set; }

    [ForeignKey("Candidate")]
    public int? HiringManagerId { get; set; }
}

File: RecruiterLite/ClientApp/src/app/models/Company.ts
import {Candidate} from "./Candidate";

export interface Company {
  id?: number
  companyName: string
  phoneNumber: string
  streetAddress: string
  postCode: string
  county: string
  country: string
  hiringManagers?: Candidate[]
}
export interface PaginatedCompanies {
  pageIndex: number,
  pageSize: number,
  count: number,
  search?: string,
  data: Company[];
}

